runtime/src/main/cpp/Memory.cpp

```diff
@@ -897,14 +931,30 @@ void freeAggregatingFrozenContainer(ContainerHeader* container) {
 
 void runDeallocationHooks(ContainerHeader* container) {
   ObjHeader* obj = reinterpret_cast<ObjHeader*>(container + 1);
-
   for (int index = 0; index < container->objectCount(); index++) {
-    if (obj->has_meta_object()) {
-      ObjHeader::destroyMetaObject(&obj->typeInfoOrMeta_);
+    if (KonanNeedDebugInfo && g_checkLeaks && (obj->type_info()->flags_ & TF_LEAK_DETECTOR_CANDIDATE) != 0) {
```

Comment:
`g_checkLeaks` here is typically `true`, so it may make sense to check type info flag first to reduce the performance cost of the entire check.

Reply:
Also, if the object doesn't have a meta object, then this entire `if` doesn't seem to have any sense.
So it probably can be moved under `if (obj->has_meta_object()) {` below.